api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -1000,23 +1000,13 @@ public void exitFromCare(Patient patient, Date dateExited, Concept reasonForExit
 	 */
 	private void saveReasonForExitObs(Patient patient, Date exitDate, Concept cause) throws APIException {
 		
-		if (patient == null) {
-			throw new APIException("Patient.null", (Object[]) null);
-		}
-		if (exitDate == null) {
-			throw new APIException("Patient.exit.date.null", (Object[]) null);
-		}
-		if (cause == null) {
-			throw new APIException("Patient.cause.null", (Object[]) null);
-		}
-		
```

Comment:
Did you intentionally remove all this a part of the changes ?

Reply:
> @ayush2297 thanks for the work on this ticket,,i just have a few questions to ask.
> 
> 1.According to the ticket description aren't the tests supposed to go in` PatientServiceTest` where existing tests already are?
> 2.why are we opting for mockito here ?

- https://issues.openmrs.org/browse/TRUNK-5069
- there was a discussion in the comments section, wherein i was told i can write unit tests using mocks
- PatientServiceTest seemed to be full of integration tests and introducing mocks required for unit tests almost broke all of the integration tests in this test class.. 
- so i created a separate class for remaining tests and followed unit testing approach in it..

================================================
> Did you intentionally remove all this a part of the changes ?

okay... so let me explain : 
- this code fragment of three if blocks was a part of saveReasonForExitObs() (which is a private function with only one use anywhere, ie. in exitFromCare() )
- i checked that saveReasonForExitObs() wasn't being called from any other function body
- the SAME  three if blocks were already present exitFromCare()
- so if that null value existed, it would never ever get thrown from the  saveReasonForExitObs(), it will always get thrown from exitFromCare()
- meaning it was an unreachable code (dead code)
- hence i removed it (since it was a unreachable dead code and untestable code)

Reply:
Lets please first get your tests merged before making any change to existing code. Lets try to make the commits small, each with a specific purpose. First lets increase test coverage. Once we have that you can make a small commit that removes the unreachable code with a commit message that explains that well. 👍

Reply:
okay... so should i put that code back in the private function and then commit again without writing tests for it (since it can't get tested)?

Reply:
@ayush2297 well, you can also create a commit with just that deletion of the unreachable code. Very nice observation! Thank you :) Once we have it merged you can push some commits were you add unit tests into https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/api/impl/PatientServiceImplTest.java